### DISCUSSION TITLE
langchain to avoid vendor lock in

### DISCUSSION THREAD

**Discussion by belda** [TheR1D/shell_gpt#430] [Functions]
Hi  have you considered using https://github.com/langchain-ai/langchain ?

It would be a nice addition to easily replace the LLM driving it and soon we can run completely opensource model locally.

**Comment by TheR1D** [comment]
It is possible to run other LLMs locally using [LocalAI](https://github.com/mudler/LocalAI) or other OpenAI compatible local backends. In the config file we can point `sgpt` requests to **localhost** by changing `OPENAI_BASE_URL`.

**Reply by belda** [reply]
Omg, great to find out about LocalAI!
